openmetadata-docs/content/v1.2.0-SNAPSHOT/connectors/database/couchbase/yaml.md

```diff
@@ -0,0 +1,227 @@
+the steps to create a YAML configuration able to connect to the source,
+process the Entities if needed, and reach the OpenMetadata server.
+
+The workflow is modeled around the following
+[JSON Schema](https://github.com/open-metadata/OpenMetadata/blob/main/openmetadata-spec/src/main/resources/json/schema/metadataIngestion/workflow.json)
+
+### 1. Define the YAML Config
+
+This is a sample config for Couchbase:
+
+{% codePreview %}
+
+{% codeInfoContainer %}
+
+#### Source Configuration - Service Connection
+
+{% codeInfo srNumber=1 %}
+
+**username**: Username to connect to Couchbase.
+
+{% /codeInfo %}
+
+{% codeInfo srNumber=2 %}
+
+**password**: Password to connect to Couchbase.
+
+{% /codeInfo %}
+
+{% codeInfo srNumber=3 %}
+
+**endpoint**: If couchbase is hosted on cloud then the endpoint parameter specifies the connection string and if you are using couchbase server then the endpoint parameter specifies hostname of the Couchbase. This should be specified as a string in the format `hostname` or `xyz.cloud.couchbase.com`. E.g., `localhost`.
+
+{% /codeInfo %}
+
+{% codeInfo srNumber=4 %}
+
+**bucketName**: Optional name to give to the bucket name in OpenMetadata. If left blank, we will ingest all the bucket names.
+
+{% /codeInfo %}
+
+#### Source Configuration - Source Config
+
+{% codeInfo srNumber=5 %}
+
+The `sourceConfig` is defined [here](https://github.com/open-metadata/OpenMetadata/blob/main/openmetadata-spec/src/main/resources/json/schema/metadataIngestion/databaseServiceMetadataPipeline.json):
+
+**markDeletedTables**: To flag tables as soft-deleted if they are not present anymore in the source system.
+
+**includeTables**: true or false, to ingest table data. Default is true.
+
+**includeViews**: true or false, to ingest views definitions.
+
+**databaseFilterPattern**, **schemaFilterPattern**, **tableFilternPattern**: Note that the filter supports regex as include or exclude. You can find examples [here](/connectors/ingestion/workflows/metadata/filter-patterns/database)
+
+{% /codeInfo %}
+
+#### Sink Configuration
+
+{% codeInfo srNumber=6 %}
+
+To send the metadata to OpenMetadata, it needs to be specified as `type: metadata-rest`.
+
+{% /codeInfo %}
+
+{% partial file="/v1.2.0/connectors/workflow-config.md" /%}
+
+#### Advanced Configuration
+
+{% codeInfo srNumber=7 %}
+
+**Connection Options (Optional)**: Enter the details for any additional connection options that can be sent to Athena during the connection. These details must be added as Key-Value pairs.
+
+{% /codeInfo %}
+
+
+{% /codeInfoContainer %}
+
+{% codeBlock fileName="filename.yaml" %}
+
+```yaml
+source:
+  type: couchbase
+  serviceName: local_couchbase
+  serviceConnection:
+    config:
+      type: Couchbase
+      connectionDetails:
+```
+```yaml {% srNumber=1 %}
+        username: username
+```
+```yaml {% srNumber=2 %}
+        password: password
+```
+```yaml {% srNumber=3 %}
+        endpoint: localhost
+```
```

Comment:
Can you validate this yaml, I don't see the `connectionDetails` field in couchbase connection json schema